synapse/lib/ast.py

```diff
@@ -4105,7 +4108,11 @@ async def run(self, runt, genr):
                         valu, _ = prop.type.norm(valu)
                         valu = prop.type.merge(oldv, valu)
 
-                await node.set(name, valu)
+                if node.form.isrunt:
+                    await node.set(name, valu)
+                else:
+                    async with runt.view.getNodeEditor(node, runt=runt) as protonode:
```

Comment:
:eyes: 
Do we have a performance hit for making a editor per node?

Reply:
So this is following the convention of the edit methods on a real node grabbing an editor and using that to perform the edit for consistency. There are probably other cases as well where we could reuse the editor, but I think creating the protonodes is probably the more expensive part. I'm not sure if there's a good way to quickly generate the nodeedits in the single edit case without duplicating a lot of the code in the ProtoNode.

Reply:
Sorry about this - the comment is misleading. We had previously made a per-set editor; we just do it in the ast now.